src/main.rs

```diff
@@ -42,7 +53,7 @@ fn bind_socket(addr: SocketAddr) -> UdpSocket {
 }
 
 /// Save DHT keys to a binary file.
-fn save_keys(keys_file: String, pk: PublicKey, sk: SecretKey) {
+fn save_keys(keys_file: &String, pk: PublicKey, sk: SecretKey) {
```

Comment:
BTW, it does not have to accept PK in args

Reply:
It has to since this function saves it as well.

Reply:
You may get it from sk.

Reply:
Not get but compute. It's computationally expensive operation so it's better to reuse keys where we can.

Reply:
> expensive operation

You do it only once. It does not affect on the speed of the server at all.

Reply:
But why do it at all when I can just pass it as a parameter? As easy as that. This function even doesn't have too many arguments.